packages/rehype-mathjax/lib/core.js

```diff
@@ -2,19 +2,35 @@ const visit = require('unist-util-visit')
 
 module.exports = createPlugin
 
-function createPlugin(displayName, createRenderer) {
+/* To do next major: Remove `chtml` and `browser` flags once all the options use
+the same format */
+
+function createPlugin(displayName, createRenderer, chtml, browser) {
```

Comment:
You use two new arguments `chtml` `browser` to declare what feature is enabled, don't you?
If so, I prefer to use `features` argument like

```js
// Typescript: 
// type Feature = "chtml" | "browser"
// type Features = Array<Feature>
function createPlugin(displayName, createRenderer, features) {
// blah blah blah ...
  if (features.includes("chtml") && ...)
```

I feel the above code is simpler than.

Reply:
I agree that a single flag parameter could be used. However, the flags I have currently do not correspond to supported features. Rather, they are only used internally to know how to "normalize" the input format, without increasing the bundle size.

https://github.com/remarkjs/remark-math/pull/54/files#diff-58df6482f24a55ac377b4e8eed3a71947c97309172516b6c10f0b647dc54f8f3R20-R25

That is, I rearrange the options based on the current client interface, whereby

* The options for the browser plugin are those of the TeX input processor,
* The options for the non-browser plugins are those of the corresponding output processor (CHTML or SVG).

This PR extends the options with the key `tex` to support configuring the TeX input processor for the non-browser plugins.
Ultimately, the flags and normalization should be removed when the options format is changed to match that of MathJax in the next major version.